From 27a329cfc69f3df40f346a6ea6388bc9e43f41fc Mon Sep 17 00:00:00 2001
From: Zebediah Figura <zfigura@codeweavers.com>
Date: Mon, 13 Aug 2018 15:04:04 -0500
Subject: [PATCH] winex11.drv: fs hack, don't try to resize child windows

Since they don't have a whole_window this will cause X errors.
---
 dlls/winex11.drv/window.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index 49125b8cb80..09abedf08ec 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -2375,7 +2375,8 @@ void CDECL X11DRV_WindowPosChanging( HWND hwnd, HWND insert_after, UINT swp_flag
         POINT p = fs_hack_real_mode();
         TRACE("Enabling fs hack, resizing the window to (%u,%u)-(%u,%u)\n", tl.x, tl.y, p.x, p.y);
         data->fs_hack = TRUE;
-        XMoveResizeWindow(data->display, data->whole_window, tl.x, tl.y, p.x, p.y);
+        if(data->whole_window)
+            XMoveResizeWindow(data->display, data->whole_window, tl.x, tl.y, p.x, p.y);
         if(data->client_window)
             XMoveResizeWindow(data->display, data->client_window, 0, 0, p.x, p.y);
     }else if(data->fs_hack &&
@@ -2384,10 +2385,11 @@ void CDECL X11DRV_WindowPosChanging( HWND hwnd, HWND insert_after, UINT swp_flag
                 window_rect->bottom - window_rect->top)){
         TRACE("Disabling fs hack\n");
         data->fs_hack = FALSE;
-        XMoveResizeWindow(data->display, data->whole_window,
-                window_rect->left, window_rect->top,
-                window_rect->right - window_rect->left,
-                window_rect->bottom - window_rect->top);
+        if(data->whole_window)
+            XMoveResizeWindow(data->display, data->whole_window,
+                    window_rect->left, window_rect->top,
+                    window_rect->right - window_rect->left,
+                    window_rect->bottom - window_rect->top);
         if(data->client_window){
             XMoveResizeWindow(data->display, data->client_window,
                     data->client_rect.left, data->client_rect.top,
